analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocations.cs

```diff
@@ -136,11 +137,73 @@ End Sub
         }
 
         [DataTestMethod]
-        [DataRow("DoSomething();")]
-        [DataRow("this.DoSomething();")]
-        [DataRow("(this).DoSomething();")]
-        [DataRow("(((this))).DoSomething();")]
-        public void Invocation_InstanceMethodCallDoesNotClearFieldOnThis(string invocation)
+        [DataRow("Initialize();")]
+        [DataRow("this.Initialize();")]
+        [DataRow("(this).Initialize();")]
+        [DataRow("(((this))).Initialize();")]
+        [DataRow("((IDisposable)this).Dispose();")]
+        [DataRow("((IDisposable)(object)this).Dispose();")]
+        [DataRow("this.SomeExtensionOnSample();")]
+        [DataRow("Extensions.SomeExtensionOnSample(this);")]
+        [DataRow("this.SomeExtensionOnObject();")]
+        [DataRow("Extensions.SomeExtensionOnObject(this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((IDisposable)this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((object)(IDisposable)this);")]
+        [DataRow("((object)(IDisposable)this).SomeExtensionOnObject();")]
+        public void Invocation_InstanceMethodCallDoesClearFieldOnThis(string invocation)
+        {
+            var code = $@"
+using System;
+namespace N
+{{
+    public class Sample: IDisposable
+    {{
+        object field1 = null;
+        object field2 = null;
+        static object staticField1 = null;
+        static object staticField2 = null;
+
+        void CallToMethodsShouldResetFieldConstraints()
```

Comment:
Purpose is in the test name already
```suggestion
        void Main()
```